File: 2.6.cpp
#include <iostream>
#include <math.h>
#include <stdio.h>

using namespace std;

int main()
{
    double n = 1.0;
    double molecule = 1.0;
    double denominator;
    double c;
    double additional;

    while (true)
    {
        // if (n == 100)
        // {
        //     break;
        // }
        denominator = log2(n);
        if (denominator)
        {
            c = molecule / denominator;
            cout << c << endl;
        }
        n += 1.0;
        molecule += 1.0 / n;
    }
}

File: 2.6.py
import math

molecule = 1
n = 1
c = 0

while True:
    denominator = math.log2(n)
    if denominator:
        c1 = molecule / denominator
        if c < c1:
            print("exceded value = {}".format(c1))
        if c == c1:
            print("stop decreasing value = {}".format(c))
        print(c1)
        c = c1
    n += 1
    molecule += 1 / n
